assets/js/modules/ads/components/common/WooCommerceRedirectModal.test.js

```diff
@@ -167,6 +226,86 @@ describe( 'WooCommerceRedirectModal', () => {
 		expect( onBeforeSetupCallback ).toHaveBeenCalled();
 	} );
 
+	it( 'clicking "Continue with Site Kit" should trigger the correct internal tracking event when only WooCommerce is active', async () => {
+		registry.dispatch( MODULES_ADS ).receiveModuleData( {
+			plugins: {
+				[ PLUGINS.WOOCOMMERCE ]: {
+					active: true,
+				},
+				[ PLUGINS.GOOGLE_FOR_WOOCOMMERCE ]: {
+					active: false,
+					adsConnected: false,
+				},
+			},
+		} );
+
+		fetchMock.postOnce( moduleActivationEndpoint, {
+			body: { success: true },
+		} );
+		fetchMock.getOnce( userAuthenticationEndpoint, {
+			body: { needsReauthentication: false },
+		} );
```

Comment:
I think this should be something we can avoid with `provideUserAuthentication`. We should only mock fetch requests that are really needed, or the subject of the test (e.g.  a resolver that fetches).

Reply:
I borrowed this from existing tests in the same file, I can explore some alternatives

Reply:
@aaemnnosttv it appears - for reasons I am not 100% certain of - that this is required. Even using `provideUserAuthentication( registry )` either within `beforeEach()` or per scenario still produces an error:

```
    Expected mock function not to be called but it was called with:
    ["Google Site Kit API Error", "method:GET", "datapoint:authentication", "type:core", "identifier:user", "error:\"fetch-mock: No fallback response defined for GET to /google-site-kit/v1/core/user/data/authentication?_locale=user\""]
```

Reply:
Ah yes, I remember now. It's because we toggle the module's activation so it triggers a refetch of the authentication data. This is how we know if the user needs to go through oauth due to required scopes changing when a module is activated.
https://github.com/google/site-kit-wp/blob/a26c2518f980353745bff2c1f46f6098f70ce20a/assets/js/googlesitekit/modules/datastore/modules.js#L310-L320

All good 👍